Build fails on Opensuse

# Environment
OS: openSUSE Tumbleweed x86_64
Kernel: 5.17.9-1-default
gcc: 12
g++: 12
# Logs
```shell
❯ bazelisk build //...

WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
INFO: Reading 'startup' options from /home/lyqingye/workspace/github/risc0/.bazelrc: --windows_enable_symlinks
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=183
INFO: Reading rc options for 'build' from /home/lyqingye/workspace/github/risc0/.bazelrc:
  'build' options: -c opt --announce_rc --enable_runfiles --nobuild_python_zip --incompatible_enable_cc_toolchain_resolution --flag_alias=accel=//risc0/zkp/accel:flag --//bazel/rules/clang_format:config=//:.clang-format --@rules_rust//:rustfmt.toml=//bazel/rules/rust:rustfmt.toml
INFO: Analyzed 206 targets (0 packages loaded, 0 targets configured).
INFO: Found 206 targets...
ERROR: /home/lyqingye/workspace/github/risc0/risc0/zkp/verify/BUILD.bazel:5:11: Compiling risc0/zkp/verify/taps.cpp failed: (Exit 1): gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 21 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from risc0/zkp/verify/taps.cpp:15:
./risc0/zkp/verify/taps.h:38:18: error: 'size_t' does not name a type
   38 | static constexpr size_t kNumRegisterGroups = 3;
      |                  ^~~~~~
./risc0/zkp/verify/taps.h:21:1: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
   20 | #include <vector>
  +++ |+#include <cstddef>
   21 | 
./risc0/zkp/verify/taps.h:72:3: error: 'size_t' does not name a type
   72 |   size_t offset() const { return data_->offset; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:72:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:73:3: error: 'size_t' does not name a type
   73 |   size_t back() const { return data_->back; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:73:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:74:3: error: 'size_t' does not name a type
   74 |   size_t comboID() const { return data_->combo; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:74:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:102:3: error: 'size_t' does not name a type
  102 |   size_t offset() const { return data_->offset; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:102:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:103:3: error: 'size_t' does not name a type
  103 |   size_t comboID() const { return data_->combo; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:103:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:110:3: error: 'size_t' does not name a type
  110 |   size_t operator[](size_t i) { return data_[i].back; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:110:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:149:43: error: 'size_t' has not been declared
  149 |   ComboRef(const detail::ComboData* data, size_t id) : data_(data), id_(id) {}
      |                                           ^~~~~~
./risc0/zkp/verify/taps.h:150:3: error: 'size_t' does not name a type
  150 |   size_t id() const { return id_; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:150:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:151:3: error: 'size_t' does not name a type
  151 |   size_t size() const { return data_->offsets[id_ + 1] - data_->offsets[id_]; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:151:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:157:3: error: 'size_t' does not name a type
  157 |   size_t id_;
      |   ^~~~~~
./risc0/zkp/verify/taps.h:157:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h: In constructor 'risc0::ComboRef::ComboRef(const risc0::detail::ComboData*, int)':
./risc0/zkp/verify/taps.h:149:69: error: class 'risc0::ComboRef' does not have any field named 'id_'
  149 |   ComboRef(const detail::ComboData* data, size_t id) : data_(data), id_(id) {}
      |                                                                     ^~~
./risc0/zkp/verify/taps.h: In member function 'const uint16_t* risc0::ComboRef::begin() const':
./risc0/zkp/verify/taps.h:152:71: error: 'id_' was not declared in this scope
  152 |   const uint16_t* begin() const { return data_->taps + data_->offsets[id_]; }
      |                                                                       ^~~
./risc0/zkp/verify/taps.h: In member function 'const uint16_t* risc0::ComboRef::end() const':
./risc0/zkp/verify/taps.h:153:69: error: 'id_' was not declared in this scope
  153 |   const uint16_t* end() const { return data_->taps + data_->offsets[id_ + 1]; }
      |                                                                     ^~~
./risc0/zkp/verify/taps.h: At global scope:
./risc0/zkp/verify/taps.h:162:48: error: 'size_t' has not been declared
  162 |   ComboIterator(const detail::ComboData* data, size_t id) : data_(data), id_(id) {}
      |                                                ^~~~~~
./risc0/zkp/verify/taps.h:171:3: error: 'size_t' does not name a type
  171 |   size_t id_;
      |   ^~~~~~
./risc0/zkp/verify/taps.h:171:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h: In constructor 'risc0::ComboIterator::ComboIterator(const risc0::detail::ComboData*, int)':
./risc0/zkp/verify/taps.h:162:74: error: class 'risc0::ComboIterator' does not have any field named 'id_'
  162 |   ComboIterator(const detail::ComboData* data, size_t id) : data_(data), id_(id) {}
      |                                                                          ^~~
./risc0/zkp/verify/taps.h: In member function 'risc0::ComboRef risc0::ComboIterator::operator*() const':
./risc0/zkp/verify/taps.h:163:55: error: 'id_' was not declared in this scope
  163 |   ComboRef operator*() const { return ComboRef(data_, id_); }
      |                                                       ^~~
./risc0/zkp/verify/taps.h: In member function 'risc0::ComboIterator& risc0::ComboIterator::operator++()':
./risc0/zkp/verify/taps.h:165:5: error: 'id_' was not declared in this scope
  165 |     id_++;
      |     ^~~
./risc0/zkp/verify/taps.h: At global scope:
./risc0/zkp/verify/taps.h:178:3: error: 'size_t' does not name a type
  178 |   size_t groupBegin[kNumRegisterGroups + 1];
      |   ^~~~~~
./risc0/zkp/verify/taps.h:178:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:202:3: error: 'size_t' does not name a type
  202 |   size_t tapsSize() { return data_->groupBegin[kNumRegisterGroups]; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:202:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:226:3: error: 'size_t' does not name a type
  226 |   size_t groupSize(RegisterGroup group) {
      |   ^~~~~~
./risc0/zkp/verify/taps.h:226:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:232:3: error: 'size_t' does not name a type
  232 |   size_t combosSize() const { return data_->combos.count; }
      |   ^~~~~~
./risc0/zkp/verify/taps.h:232:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
./risc0/zkp/verify/taps.h:239:21: error: 'size_t' has not been declared
  239 |   ComboRef getCombo(size_t id) { return ComboRef(&data_->combos, id); }
      |                     ^~~~~~
./risc0/zkp/verify/taps.h: In member function 'risc0::IteratorRange<risc0::TapIterator> risc0::TapSetRef::taps()':
./risc0/zkp/verify/taps.h:206:60: error: 'const struct risc0::detail::TapSetData' has no member named 'groupBegin'
  206 |                                       data_->taps + data_->groupBegin[kNumRegisterGroups]);
      |                                                            ^~~~~~~~~~
./risc0/zkp/verify/taps.h:206:71: error: 'kNumRegisterGroups' was not declared in this scope; did you mean 'RegisterGroup'?
  206 |                                       data_->taps + data_->groupBegin[kNumRegisterGroups]);
      |                                                                       ^~~~~~~~~~~~~~~~~~
      |                                                                       RegisterGroup
./risc0/zkp/verify/taps.h: In member function 'risc0::IteratorRange<risc0::RegIterator> risc0::TapSetRef::regs()':
./risc0/zkp/verify/taps.h:211:60: error: 'const struct risc0::detail::TapSetData' has no member named 'groupBegin'
  211 |                                       data_->taps + data_->groupBegin[kNumRegisterGroups]);
      |                                                            ^~~~~~~~~~
./risc0/zkp/verify/taps.h:211:71: error: 'kNumRegisterGroups' was not declared in this scope; did you mean 'RegisterGroup'?
  211 |                                       data_->taps + data_->groupBegin[kNumRegisterGroups]);
      |                                                                       ^~~~~~~~~~~~~~~~~~
      |                                                                       RegisterGroup
./risc0/zkp/verify/taps.h: In member function 'risc0::IteratorRange<risc0::TapIterator> risc0::TapSetRef::groupTaps(risc0::RegisterGroup)':
./risc0/zkp/verify/taps.h:215:5: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
  215 |     size_t groupID = static_cast<size_t>(group);
      |     ^~~~~~
      |     std::size_t
In file included from /usr/include/c++/12/cstdint:38,
                 from ./risc0/zkp/verify/taps.h:17:
/usr/include/c++/12/x86_64-suse-linux/bits/c++config.h:2535:33: note: 'std::size_t' declared here
 2535 |   typedef __SIZE_TYPE__         size_t;
      |                                 ^~~~~~
./risc0/zkp/verify/taps.h:216:60: error: 'const struct risc0::detail::TapSetData' has no member named 'groupBegin'
  216 |     return IteratorRange<TapIterator>(data_->taps + data_->groupBegin[groupID],
      |                                                            ^~~~~~~~~~
./risc0/zkp/verify/taps.h:216:71: error: 'groupID' was not declared in this scope; did you mean 'group'?
  216 |     return IteratorRange<TapIterator>(data_->taps + data_->groupBegin[groupID],
      |                                                                       ^~~~~~~
      |                                                                       group
./risc0/zkp/verify/taps.h:217:60: error: 'const struct risc0::detail::TapSetData' has no member named 'groupBegin'
  217 |                                       data_->taps + data_->groupBegin[groupID + 1]);
      |                                                            ^~~~~~~~~~
./risc0/zkp/verify/taps.h: In member function 'risc0::IteratorRange<risc0::RegIterator> risc0::TapSetRef::groupRegs(risc0::RegisterGroup)':
./risc0/zkp/verify/taps.h:221:5: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
  221 |     size_t groupID = static_cast<size_t>(group);
      |     ^~~~~~
      |     std::size_t
/usr/include/c++/12/x86_64-suse-linux/bits/c++config.h:2535:33: note: 'std::size_t' declared here
 2535 |   typedef __SIZE_TYPE__         size_t;
      |                                 ^~~~~~
./risc0/zkp/verify/taps.h:222:60: error: 'const struct risc0::detail::TapSetData' has no member named 'groupBegin'
  222 |     return IteratorRange<RegIterator>(data_->taps + data_->groupBegin[groupID],
      |                                                            ^~~~~~~~~~
./risc0/zkp/verify/taps.h:222:71: error: 'groupID' was not declared in this scope; did you mean 'group'?
  222 |     return IteratorRange<RegIterator>(data_->taps + data_->groupBegin[groupID],
      |                                                                       ^~~~~~~
      |                                                                       group
./risc0/zkp/verify/taps.h:223:60: error: 'const struct risc0::detail::TapSetData' has no member named 'groupBegin'
  223 |                                       data_->taps + data_->groupBegin[groupID + 1]);
      |                                                            ^~~~~~~~~~
./risc0/zkp/verify/taps.h: At global scope:
./risc0/zkp/verify/taps.h:242:22: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
  242 | using Reg = std::set<size_t>;
      |                      ^~~~~~
      |                      std::size_t
/usr/include/c++/12/x86_64-suse-linux/bits/c++config.h:2535:33: note: 'std::size_t' declared here
 2535 |   typedef __SIZE_TYPE__         size_t;
      |                                 ^~~~~~
./risc0/zkp/verify/taps.h:242:28: error: template argument 1 is invalid
  242 | using Reg = std::set<size_t>;
      |                            ^
./risc0/zkp/verify/taps.h:242:28: error: template argument 2 is invalid
./risc0/zkp/verify/taps.h:242:28: error: template argument 3 is invalid
./risc0/zkp/verify/taps.h:243:24: error: 'size_t' was not declared in this scope; did you mean 'std::size_t'?
  243 | using Group = std::map<size_t, Reg>;
      |                        ^~~~~~
      |                        std::size_t
/usr/include/c++/12/x86_64-suse-linux/bits/c++config.h:2535:33: note: 'std::size_t' declared here
 2535 |   typedef __SIZE_TYPE__         size_t;
      |                                 ^~~~~~
./risc0/zkp/verify/taps.h:243:32: error: 'Reg' was not declared in this scope
  243 | using Group = std::map<size_t, Reg>;
      |                                ^~~
./risc0/zkp/verify/taps.h:243:35: error: template argument 1 is invalid
  243 | using Group = std::map<size_t, Reg>;
      |                                   ^
./risc0/zkp/verify/taps.h:243:35: error: template argument 2 is invalid
./risc0/zkp/verify/taps.h:243:35: error: template argument 3 is invalid
./risc0/zkp/verify/taps.h:243:35: error: template argument 4 is invalid
./risc0/zkp/verify/taps.h:244:37: error: 'Group' was not declared in this scope
  244 | using All = std::map<RegisterGroup, Group>;
      |                                     ^~~~~
./risc0/zkp/verify/taps.h:244:42: error: template argument 2 is invalid
  244 | using All = std::map<RegisterGroup, Group>;
      |                                          ^
./risc0/zkp/verify/taps.h:244:42: error: template argument 4 is invalid
./risc0/zkp/verify/taps.h:254:16: error: 'All' does not name a type
  254 |   TapSet(const All& all);
      |                ^~~
risc0/zkp/verify/taps.cpp:25:22: error: 'All' does not name a type
   25 | TapSet::TapSet(const All& all) {
      |                      ^~~
risc0/zkp/verify/taps.cpp: In constructor 'risc0::TapSet::TapSet(const int&)':
risc0/zkp/verify/taps.cpp:26:15: error: 'Reg' was not declared in this scope
   26 |   std::vector<Reg> combos;
      |               ^~~
risc0/zkp/verify/taps.cpp:26:18: error: template argument 1 is invalid
   26 |   std::vector<Reg> combos;
      |                  ^
risc0/zkp/verify/taps.cpp:26:18: error: template argument 2 is invalid
risc0/zkp/verify/taps.cpp:27:23: error: template argument 3 is invalid
   27 |   std::map<Reg, size_t> combosToID;
      |                       ^
risc0/zkp/verify/taps.cpp:27:23: error: template argument 4 is invalid
risc0/zkp/verify/taps.cpp:30:10: error: request for member 'push_back' in 'combos', which is of non-class type 'int'
   30 |   combos.push_back(self);
      |          ^~~~~~~~~
risc0/zkp/verify/taps.cpp:31:14: error: request for member 'insert' in 'combosToID', which is of non-class type 'int'
   31 |   combosToID.insert(std::make_pair(self, 0));
      |              ^~~~~~
risc0/zkp/verify/taps.cpp:33:38: error: 'kNumRegisterGroups' was not declared in this scope; did you mean 'RegisterGroup'?
   33 |   for (size_t groupID = 0; groupID < kNumRegisterGroups; groupID++) {
      |                                      ^~~~~~~~~~~~~~~~~~
      |                                      RegisterGroup
In file included from risc0/zkp/verify/taps.cpp:17:
risc0/zkp/verify/taps.cpp:36:17: error: request for member 'count' in 'all', which is of non-class type 'const int'
   36 |     REQUIRE(all.count(group));
      |                 ^~~~~
./risc0/core/util.h:25:11: note: in definition of macro 'REQUIRE'
   25 |     if (!(x)) {                                                                                    \
      |           ^
risc0/zkp/verify/taps.cpp:38:11: error: 'struct risc0::detail::TapSetData' has no member named 'groupBegin'
   38 |     data_.groupBegin[groupID] = taps_.size();
      |           ^~~~~~~~~~
risc0/zkp/verify/taps.cpp:40:11: error: 'Group' does not name a type
   40 |     const Group& regs = all.at(group);
      |           ^~~~~
risc0/zkp/verify/taps.cpp:42:23: error: 'regs' was not declared in this scope
   42 |     size_t regCount = regs.rbegin()->first + 1;
      |                       ^~~~
risc0/zkp/verify/taps.cpp:47:10: error: expected ';' before 'combo'
   47 |       Reg combo = regs.at(reg);
      |          ^~~~~~
      |          ;
risc0/zkp/verify/taps.cpp:48:15: error: 'combo' was not declared in this scope; did you mean 'combos'?
   48 |       REQUIRE(combo.size() > 0);
      |               ^~~~~
./risc0/core/util.h:25:11: note: in definition of macro 'REQUIRE'
   25 |     if (!(x)) {                                                                                    \
      |           ^
risc0/zkp/verify/taps.cpp:50:23: error: request for member 'count' in 'combosToID', which is of non-class type 'int'
   50 |       if (!combosToID.count(combo)) {
      |                       ^~~~~
risc0/zkp/verify/taps.cpp:50:29: error: 'combo' was not declared in this scope; did you mean 'combos'?
   50 |       if (!combosToID.count(combo)) {
      |                             ^~~~~
      |                             combos
risc0/zkp/verify/taps.cpp:51:36: error: request for member 'size' in 'combos', which is of non-class type 'int'
   51 |         combosToID[combo] = combos.size();
      |                                    ^~~~
risc0/zkp/verify/taps.cpp:52:16: error: request for member 'push_back' in 'combos', which is of non-class type 'int'
   52 |         combos.push_back(combo);
      |                ^~~~~~~~~
risc0/zkp/verify/taps.cpp:55:35: error: 'combo' was not declared in this scope; did you mean 'combos'?
   55 |       size_t comboID = combosToID[combo];
      |                                   ^~~~~
      |                                   combos
risc0/zkp/verify/taps.cpp:72:9: error: 'struct risc0::detail::TapSetData' has no member named 'groupBegin'
   72 |   data_.groupBegin[kNumRegisterGroups] = taps_.size();
      |         ^~~~~~~~~~
risc0/zkp/verify/taps.cpp:72:20: error: 'kNumRegisterGroups' was not declared in this scope; did you mean 'RegisterGroup'?
   72 |   data_.groupBegin[kNumRegisterGroups] = taps_.size();
      |                    ^~~~~~~~~~~~~~~~~~
      |                    RegisterGroup
risc0/zkp/verify/taps.cpp:75:21: error: 'begin' was not declared in this scope; did you mean 'std::begin'?
   75 |   for (auto combo : combos) {
      |                     ^~~~~~
      |                     std::begin
In file included from /usr/include/c++/12/map:63,
                 from ./risc0/zkp/verify/taps.h:18:
/usr/include/c++/12/bits/range_access.h:114:37: note: 'std::begin' declared here
  114 |   template<typename _Tp> const _Tp* begin(const valarray<_Tp>&) noexcept;
      |                                     ^~~~~
risc0/zkp/verify/taps.cpp:75:21: error: 'end' was not declared in this scope; did you mean 'std::end'?
   75 |   for (auto combo : combos) {
      |                     ^~~~~~
      |                     std::end
/usr/include/c++/12/bits/range_access.h:116:37: note: 'std::end' declared here
  116 |   template<typename _Tp> const _Tp* end(const valarray<_Tp>&) noexcept;
      |                                     ^~~
risc0/zkp/verify/taps.cpp:85:31: error: request for member 'size' in 'combos', which is of non-class type 'int'
   85 |   data_.combos.count = combos.size();
      |                               ^~~~
INFO: Elapsed time: 0.606s, Critical Path: 0.43s
INFO: 40 processes: 35 internal, 5 linux-sandbox.
FAILED: Build did NOT complete successfully

```

--- a/risc0/zkp/verify/taps.h
+++ b/risc0/zkp/verify/taps.h
@@ -14,6 +14,7 @@
 
 #pragma once
 
+#include <cstddef>
 #include <cstdint>
 #include <map>
 #include <set>

